ddtrace/_logger.py

```diff
@@ -0,0 +1,50 @@
+import logging
+from logging.handlers import RotatingFileHandler
+import os
+
+from ddtrace.internal.utils.formats import asbool
+
+
+DEFAULT_FILE_SIZE_BYTES = 15 << 20  # 15 MB
+
+
+def configure_ddtrace_logger():
+    # type: () -> None
+    """Configures ddtrace log levels and file paths.
+    By default, when debug hasn't been enabled and a file hasn't been specified:
+        - Inherit from the root logger in the logging module
+
+    In debug mode:
+        - Propagate logs up so logs show up in the application logs when a file path wasn't provided
+        - Logs are routed to a file when DD_TRACE_LOG_FILE is specified
+        - Child loggers inherit from the parent ddtrace logger
+
+    Note: The ddtrace-run logs under commands/ddtrace-run do not follow DD_TRACE_LOG_FILE
+        if DD_TRACE_DEBUG is enabled.
+
+    """
+    debug_enabled = asbool(os.environ.get("DD_TRACE_DEBUG", "false"))
+    log_path = os.environ.get("DD_TRACE_LOG_FILE")
+    max_file_bytes = int(os.environ.get("DD_TRACE_FILE_SIZE_BYTES", DEFAULT_FILE_SIZE_BYTES))
+    num_backup = 1
+    log_format = "%(asctime)s %(levelname)s [%(name)s] [%(filename)s:%(lineno)d] - %(message)s"
+    log_formatter = logging.Formatter(log_format)
+
+    ddtrace_logger = logging.getLogger("ddtrace")
+
+    if debug_enabled is True:
+        ddtrace_logger.setLevel(logging.DEBUG)
+        ddtrace_logger.debug("Debug mode has been enabled for the ddtrace logger.")
+
+    if log_path is not None:
+        log_path = os.path.abspath(log_path)
+        ddtrace_file_handler = RotatingFileHandler(
+            filename=log_path, mode="a", maxBytes=max_file_bytes, backupCount=num_backup
+        )
+        if debug_enabled is False:
+            ddtrace_file_handler.setLevel(logging.WARN)
+        else:
+            ddtrace_file_handler.setLevel(logging.DEBUG)
```

Comment:
We were discussing this elsewhere but thought it good to bring into the PR here.

Right now a user would have to both set `DD_TRACE_DEBUG=true` and `DD_TRACE_LOG_FILE` to a path for this behavior.

If they instead don't set `DD_TRACE_DEBUG=true`, the log file would be configured with logging level of warning.

Being that we are introducing this as a new feature to make it easy for users to share debug logs when they are having issues, we should make that it super simple to configure. 

My vote here would be to have setting one environment variable (`DD_TRACE_LOG_FILE`) be adequate to have debug logs logged to a file. 

We can as well add now a `DD_TRACE_LOG_FILE_LEVEL` with the default value `DEBUG`. Or we can add such a configuration option based on user feedback.

Reply:
I am ok with this suggestion.

In the end we would have 3 env variables:

- `DD_TRACE_DEBUG` - whether `logging.getLogger("ddtrace").setLevel(logging.DEBUG)` is set.
- `DD_TRACE_LOG_FILE` - adds a log handler pushing logs into this file
- `DD_TRACE_LOG_FILE_LEVEL` - sets the log level to push into the file, defaults to "DEBUG"

Things that I think make this work... the log file will not log debug logs until you set the level on the root logger.

That is, doing this:

```python
import logging


root = logging.getLogger("ddtrace")

handler = logging.StreamHandler()
handler.setLevel(logging.DEBUG)
root.addHandler(handler)


assert root.isEnabledFor(logging.DEBUG) is False
root.debug("test")  # Does not get logged
```

Even though the handler is there and set to debug it does not log anything.